src/Components/Components/src/SupplyParameterFromPersistentComponentStateAttribute.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Indicates that the value for the parameter might come from persistent component state from a
+/// previous render.
+/// </summary>
+[AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
+public sealed class SupplyParameterFromPersistentComponentStateAttribute : CascadingParameterAttributeBase
```

Comment:
Would love to see this called just `PersistAttribute`, so the usage looks like:

```cs
[Persist]
public int Count { get; set; }
```

I think that will make the whole feature a lot more appealing and seem less advanced.

Reply:
If you're concerned the name is too general, we could go for something like `[ComponentPersist]`. It's not as friendly but would still be a lot more appealing than `[SupplyParameterFromPersistentComponentState]`.

Reply:
`[SupplyParameterFromState]` is my take. I want it to have the same prefix as `SupplyParameterFrom(Form|Query|State)` and so on. I am going to shun the `PersistentComponent` stuff from new APIs

Reply:
I'm going to table the naming until we do API review, as it will likely change anyway. For now it gets the ugly name until we get to API review.